src/run_releasenotes_check.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import os
+import subprocess
+from typing import Any, List
+
+from pytablewriter import MarkdownTableWriter
+
+from git.git_repository import GitRepository
+from manifests.input_manifest import InputComponentFromSource, InputManifest
+from releasenotes_check_workflow.releasenotes_check_args import ReleaseNotesCheckArgs
+from system import console
+from system.temporary_directory import TemporaryDirectory
+
+
+def main() -> int:
+    value_matrix = []  # type: List[Any]
+    args = ReleaseNotesCheckArgs()
+    console.configure(level=args.logging_level)
+    manifest = InputManifest.from_file(args.manifest)
+    writer = MarkdownTableWriter(
+        table_name=f"{manifest.build.name} CommitID(after {args.gitlogdate}) & Release Notes info",
+        headers=["Repo", "Branch", "CommitID", "Release Notes"],
+        value_matrix=value_matrix
+    )
+    with TemporaryDirectory(chdir=True) as work_dir:
+        logging.info(f"Checking out into {work_dir.name}")
+        for component in manifest.components.select():
+            logging.info(f"Checking out {component.name}")
+            repo_list = []  # type: List[Any]
+            if type(component) is InputComponentFromSource:
```

Comment:
Add a method `Component#checkout(path)` or at least a `InputComponent#checkout(path)` that returns an instance of `GitRepository`. Depending on the component type it should return OpenSearchGitRepository, OpenSearchPluginGitRepository, OpenSearchDashboardsGitRepository, etc. Each of these should have a `release_notes` property that is implemented differently depending on all this if/then/else logic you have below.  Each one of these things needs tests.